src/Controllers/DatabaseController.js

```diff
@@ -59,28 +59,37 @@ DatabaseController.prototype.validateClassName = function(className) {
   return Promise.resolve();
 };
 
-// Returns a promise for a schema object.
-// If we are provided a acceptor, then we run it on the schema.
-// If the schema isn't accepted, we reload it at most once.
-DatabaseController.prototype.loadSchema = function(acceptor = returnsTrue) {
+function _loadSchema(db) {
+  if (db._schema) {
+    return Promise.resolve(db._schema);
+  }
 
-  if (!this.schemaPromise) {
-    this.schemaPromise = this.schemaCollection().then(collection => {
-      delete this.schemaPromise;
+  if (!db.schemaPromise) {
+    db.schemaPromise = db.schemaCollection().then(collection => {
       return Schema.load(collection);
-    });
-    return this.schemaPromise;
+    }).then((res) => {
+      db._schema = res;
+      return Promise.resolve(db._schema);
+    })
   }
+  return db.schemaPromise;
+}
 
-  return this.schemaPromise.then((schema) => {
+DatabaseController.prototype.invalidateSchema = function() {
+  delete this._schema;
+  delete this.schemaPromise;
```

Comment:
I don't think you should delete this promise, as it is a outstanding request to refresh the schema, This would actually make performance worse.

this function should just be ``delete this._schema``